hw/drivers/flash/spiflash/include/spiflash/spiflash.h

```diff
@@ -33,6 +33,24 @@
 extern "C" {
 #endif
 
+/*
+ * Structure to hold typical and maximum time as stated in chip datasheet.
+ * Values are used for timeouts and are specified in micro seconds.
+ */
+struct spiflash_time_spec {
+    uint32_t typical;
+    uint32_t maximum;
+};
+
+struct spiflash_characteristics {
+    struct spiflash_time_spec tse;  /* Sector erase time (4KB) */
+    struct spiflash_time_spec tbe1; /* Block erase time (32KB) */
+    struct spiflash_time_spec tbe2; /* Block erase time (64KB) */
```

Comment:
I have not looked at spi flashes lately but these sector/block sizes seem specific. What if there is a block that is much bigger or something? Are these things generally linear and one could simply take a 32KB sector and multiply it by sector size/32KB? Or maybe we can get things from the specific spi flash itself (meaning have calls in to the specific spi flash driver code)?